### ISSUE TITLE
[Flake] Resource exhausted in spanner integration tests

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-cpp#4070]
We just got two failures, caused by the same issue:

```
Creating database and table google/cloud/spanner/testing/database_integration_test.cc:81: Failure
Value of: database
Expected: is OK
Actual: Error in non-idempotent operation CreateDatabase: The instance projects/cloud-cpp-testing-resources/instances/test-instance-5 has too many database splits to complete the operation.  Please add more nodes and try again. [RESOURCE_EXHAUSTED] [RESOURCE_EXHAUSTED]
```

https://source.cloud.google.com/results/invocations/90b7291b-013c-4abe-8e15-98785f399637
https://source.cloud.google.com/results/invocations/bde9c3a9-59a8-45a0-a478-0054bfd43ab9

**Comment by coryan** [comment]
Used this command to cleanup some databases:

```
for db in $(gcloud spanner databases list --project cloud-cpp-testing-resources --instance test-instance-5 --filter='name:db-' --format='csv(name)[no-heading]' --limit 5); do sleep 1; gcloud spanner databases delete --project cloud-cpp-testing-resources --instance test-instance-5 --quiet "${db}"; echo ${db}; done
```

**Comment by coryan** [comment]
This is (I think) related to #4033

**Comment by devjgm** [comment]
These errors say `Please add more nodes and try again`. It's a resource exhausted error. Our test instances only have a single node. I think we could probably safely make them all 2-node test instances. Perhaps we should just do that?

**Comment by coryan** [comment]
We may need to move some instances to a different region because we are close to exhausting our node quota for `us-central1` (as I recall).

If that is the path we take we would need to:

- Add more instances in a new region with 2 nodes each.
- Remove some of the existing instances (ideally with no builds running at the time, to prevent more flakes).
- Add nodes to the instances not removed in the previous step.

I suspect that will probably just buy time, eventually we will saturate two node instances too. That is a guess though.

**Comment by coryan** [comment]
Doing another cleanup (I found about 180 stale databases):

```console
for instance in $(gcloud spanner instances list --project cloud-cpp-testing-resources --filter="name:test-instance" --format='csv(name)[no-heading]'); do
  for db in $(gcloud spanner databases list --project cloud-cpp-testing-resources --instance ${instance} --filter='name:db-' --format='csv(name)[no-heading]' | grep -v 'db-2020-'); do
     echo $instance $db; sleep 1; 
     gcloud spanner databases delete --project cloud-cpp-testing-resources --instance "${instance}" --quiet "${db}";
  done
done
```

I think the fixes in #4264 and this cleanup will take care of the problem.
